src/AutoMapper/Internal/TypePair.cs

```diff
@@ -11,15 +11,17 @@ public readonly record struct TypePair(Type SourceType, Type DestinationType)
 {
     public bool IsConstructedGenericType => SourceType.IsConstructedGenericType || DestinationType.IsConstructedGenericType;
     public bool ContainsGenericParameters => SourceType.ContainsGenericParameters || DestinationType.ContainsGenericParameters;
+    public Type DestinationType { get; init; } = DestinationType.IsByRef ? DestinationType.GetElementType() : DestinationType;
```

Comment:
This is not an acceptable overhead. `TypePair` is used everywhere. This check should only appeaer only in constructor mapping code.